src/NuGet.Clients/NuGet.SolutionRestoreManager/RestoreOperationLogger.cs

```diff
@@ -159,29 +173,9 @@ public override void Log(ILogMessage logMessage)
         /// if the message needs to be logged before moving to the
         /// UI thread. 
         /// </summary>
-        public override Task LogAsync(ILogMessage logMessage)
+        public sealed override Task LogAsync(ILogMessage logMessage)
         {
-            HandleErrorsAndWarnings(logMessage);
-
-            if (DisplayMessage(logMessage.Level))
-            {
-                var verbosityLevel = GetMSBuildLevel(logMessage.Level);
-                var reportProgress = ShouldReportProgress(logMessage);
-
-                // Write to the output window if the verbosity level is high enough.
-                var showAsOutputMessage = ShouldShowMessageAsOutput(verbosityLevel);
-
-                // Avoid moving to the UI thread unless there is work to do
-                if (reportProgress || showAsOutputMessage)
-                {
-                    // Run on the UI thread
-                    return DoAsync((_, progress) =>
-                    {
-                        LogToVS(reportProgress, showAsOutputMessage, logMessage, progress);
-                    });
-                }
-            }
-
+            Log(logMessage);
             return Task.FromResult(0);
```

Comment:
1. This is an *extremely* common pattern in the current code base
2. It didn't show up as a dominant contributor in recent profiling

Based on this, I decided to treat this as a code cleanup issue which can wait for a future commit dedicated to this effort.